### PR TITLE
Configurable Consul service name separator

### LINKED ISSUE #81: Consul does not support dots in service names for DNS queries
Services names must not contain a dot (.) otherwise you could not use the Consul DNS API. For example
a Marathon App with id "/demo/hello/world" is registered by default as Consul Service "demo.hello.world"

From Consul DNS API documentation (https://www.consul.io/docs/agent/services.html):
"DNS-compliant service and tag names may contain any alpha-numeric characters, as well as dashes. Dots are not supported because Consul internally uses them to delimit service tags."

It would be nice if you could implement a different naming strategy, e.g. use dashes instead of dots.

Current workaround for us is to specify a custom consul service name.

### PR DESCRIPTION
**Pull request by dankraw**
Fixes #81 Different implementation, app -> service name conversion moved to Consul object.

Contains work from @janisz at #85 

👯 👯 👯

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -104,42 +104,41 @@ curl -X POST 'http://marathon.service.consul:8080/v2/eventSubscriptions?callback
...
+metrics-location       |                       | Graphite URL (used when metrics-target is set to graphite)
+metrics-prefix         | `default`             | Metrics prefix (resolved to `<hostname>.<app_name>` by default)
+metrics-target         | `stdout`              | Metrics destination `stdout` or `graphite` (empty string disables metrics)
+sync-enabled           | `true`                | Enable Marathon-consul scheduled sync
+sync-force             | `false`               | Force leadership-independent Marathon-consul sync (run always)
+sync-interval          | `15m0s`               | Marathon-consul sync [interval](https://golang.org/pkg/time/#Duration) **Note:** While using file configuration intervals should be provided in *nanoseconds*
+sync-leader            | `<hostname>:8080`     | Marathon cluster-wide node name (defaults to `<hostname>:8080`), the sync will run only if the node is the current [Marathon-leader](https://mesosphere.github.io/marathon/docs/rest-api.html#get-v2-leader)
+workers-pool-size      | `10`                  | Number of concurrent workers processing events
+
```
Comment (janisz) [inline README.md]: Can we minify this difference?

Comment (dankraw) [inline README.md]: Well, I didn't manually touch the file. You should know the differences better than me. :D

Comment (dankraw) [inline README.md]: OK I get it. There should be little difference. Maybe something went wrong with the rebase. :)

Comment (dankraw) [inline README.md]: done :)

File: apps/app.go [hunk 2]
```diff
@@ -45,23 +44,20 @@ func (id AppId) String() string {
 	return string(id)
 }
 
-func (id AppId) ConsulServiceName() string {
-	return strings.Replace(strings.Trim(strings.TrimSpace(id.String()), "/"), "/", ".", -1)
-}
-
 func (app *App) IsConsulApp() bool {
```
Comment (janisz) [inline apps/app.go]: If we are moving Consul related logic from here this could be also moved to consul.

File: apps/app.go [hunk 3]
```diff
@@ -45,23 +44,20 @@ func (id AppId) String() string {
...
 	return ok
 }
 
-func (app *App) ConsulServiceName() string {
-	if value, ok := app.Labels[MARATHON_CONSUL_LABEL]; ok && value != "true" {
-		value = AppId(value).ConsulServiceName()
-		if value != "" {
-			return value
-		}
+func (app *App) ConsulName() string {
```
Comment (janisz) [inline apps/app.go]: If we are moving Consul related logic from here this could be also moved to consul.

File: config/config.go [hunk 4]
```diff
@@ -71,11 +71,12 @@ func (config *Config) parseFlags() {
...
 	flag.Uint32Var(&config.Consul.RequestRetries, "consul-get-services-retry", 3, "Number of retries on failure when performing requests to Consul. Each retry uses different cached agent")
+	flag.StringVar(&config.Consul.ConsulNameSeparator, "consul-name-separator", ".", "Separator used to create default service name for Consul")
 
 	// Web
-	flag.StringVar(&config.Web.Listen, "listen", ":4000", "Accept connections at this address")
-	flag.IntVar(&config.Web.QueueSize, "events-queue-size", 1000, "Size of events queue")
-	flag.IntVar(&config.Web.WorkersCount, "workers-pool-size", 10, "Number of concurrent workers processing events")
+	flag.StringVar(&config.Web.Listen, "listen", ":4000", "accept connections at this address")
+	flag.IntVar(&config.Web.QueueSize, "events-queue-size", 1000, "size of events queue")
+	flag.IntVar(&config.Web.WorkersCount, "workers-pool-size", 10, "number of concurrent workers processing events")
```
Comment (janisz) [inline config/config.go]: Why change it to lowercase? In [last commit](https://github.com/allegro/marathon-consul/commit/6ed7737d4e5ddf82bd078005b8caa6a6b18270b5/config/config.go) I make it uppercase to be consistent with other.

File: apps/app.go (hunk 3 above)
Comment (dankraw) [inline apps/app.go]: In my opinion it should stay here as it's related to the Marathon app configuration. Here `ConsulName` returns the name that the user set for the app on the Marathon side (using it's labels or simply getting the default app-id). That's why we do the `"true"` and `""` comparison here, etc. 
On the Consul class side we have the destination service name returned with all the Consul logic applied.

File: config/config.go (hunk 4 above)
Comment (dankraw) [inline config/config.go]: Again, a rebase issue. :) Fixed, thanks!

File: apps/app.go (hunk 2 above)
Comment (dankraw) [inline apps/app.go]: Same as below, it's related to Marathon app configuration, not Consul services.

**Comment by janisz** [comment]
:+1:

**Pull request merged**